sdk/storage/azure-storage-file/apireview/azure/storage/file/directory_client.pyi

```diff
@@ -0,0 +1,40 @@
+
+from ._deserialize import deserialize_directory_properties
+from ._generated import AzureFileStorage
+from ._generated.models import HandleItem, StorageErrorException
+from ._generated.version import VERSION
+from ._parser import _datetime_to_str, _get_file_permission
+from ._polling import CloseHandles
+from ._shared.base_client import StorageAccountHostsMixin, parse_connection_str, parse_query
+from ._shared.parser import _str
+from ._shared.request_handlers import add_metadata_headers
+from ._shared.response_handlers import process_storage_error, return_response_headers
+from .file_client import FileClient
+from .models import ContentSettings, DirectoryProperties, DirectoryPropertiesPaged, HandlesPaged, NTFSAttributes, ShareProperties
+from azure.core.paging import ItemPaged
+from typing import Any, Dict, Optional, Union
+
+class DirectoryClient(StorageAccountHostsMixin):
+    snapshot: Any = ...
+    share_name: Any = ...
+    directory_path: Any = ...
+    def __init__(self, directory_url: str, share: Optional[Union[str, ShareProperties]]=..., directory_path: Optional[str]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., **kwargs: Optional[Any]) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, share: Optional[Union[str, ShareProperties]]=..., directory_path: Optional[str]=..., credential: Optional[Any]=..., **kwargs: Any) -> DirectoryClient: ...
+    def get_file_client(self, file_name: str, **kwargs: Any) -> FileClient: ...
+    def get_subdirectory_client(self, directory_name: str, **kwargs: Any) -> DirectoryClient: ...
+    def create_directory(self, metadata: Optional[Dict[str, str]]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    def delete_directory(self, timeout: Optional[int]=..., **kwargs: Any) -> None: ...
+    def list_directories_and_files(self, name_starts_with: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> ItemPaged: ...
+    def list_handles(self, recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> ItemPaged: ...
+    def close_handles(self, handle: Union[str, HandleItem]=..., recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> Any: ...
+    def get_directory_properties(self, timeout: Optional[int]=..., **kwargs: Any) -> DirectoryProperties: ...
+    def set_directory_metadata(self, metadata: Dict[str, Any], timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def set_http_headers(self, file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def create_subdirectory(self, directory_name: str, metadata: Optional[Dict[str, Any]]=..., timeout: Optional[int]=..., **kwargs: Any) -> DirectoryClient: ...
```

Comment:
I feel the distinction between `create_directory` and `create_subdirectory` is confusing....

Reply:
Keep separate: `create` and `create_subdirectory`

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/7836